### PR TITLE
Add EitherDataExtensions

### PR DESCRIPTION
**Pull request by rammba**
Hello team, thanks for implementing this library. I recently started using it and I can say that it's really helpful.

Added few conversion extensions for `EitherData`:
- ToEither
- ToEitherAsync
- ToEitherUnsafe
- ToFin
- ToTry
- ToTryAsync

### CODE REVIEW

File: LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
Comment (louthy) [inline LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs:1]: Could you add `#nullable enable` to the top of the file pls.  language-ext is being slowly migrated to support nullable references, it'd be good if new code supported it off the bat.

File: LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            input.State switch
+            {
+                EitherStatus.IsBottom => Either<L, R>.Bottom,
+                EitherStatus.IsRight => Either<L, R>.Right(input.Right),
+                EitherStatus.IsLeft => Either<L, R>.Left(input.Left),
+                _ => throw new InvalidOperationException($"Invalid {nameof(EitherStatus)} with value: {input.State}"),
+            };
+
+        public static EitherAsync<L, R> ToEitherAsync<L, R>(this EitherData<L, R> input) =>
+            input.ToEither().ToAsync();
```
Comment (louthy) [inline LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs:17]: There should also be a `ToEitherAsync` that takes `ValueTask<EitherData<L, R>>` and one that takes `Task<EitherData<L, R>>`

**Review by louthy — CHANGES_REQUESTED** [review]
There's other types that should be supported:

* `EitherUnsafe<L, R>` (allows nulls in the `L` and `R`)
* `Fin<A>` which is the same as `EitherData<Error, A>`
* `Try<A>` which is the same as `EitherData<Exception, A>`
* `TryAsync<A>` which is the same as `Task<EitherData<Exception, A>>` and `ValueTask<EitherData<Exception, A>>` 

Welcome to my world.  There's no such thing as a small change in this library ;-)

**Comment by rammba** [comment]
> There's other types that should be supported:
> 
> * `EitherUnsafe<L, R>` (allows nulls in the `L` and `R`)
> * `Fin<A>` which is the same as `EitherData<Error, A>`
> * `Try<A>` which is the same as `EitherData<Exception, A>`
> * `TryAsync<A>` which is the same as `Task<EitherData<Exception, A>>` and `ValueTask<EitherData<Exception, A>>`
> 
> Welcome to my world. There's no such thing as a small change in this library ;-)

Thanks for drawing attention on other monads, I hope my conversion is correct regarding your implementation.

File: LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+{
+    public static class EitherDataExtensions
+    {
+        public static Either<L, R> ToEither<L, R>(this EitherData<L, R> input) =>
+            input.State switch
+            {
+                EitherStatus.IsBottom => Either<L, R>.Bottom,
+                EitherStatus.IsRight => Either<L, R>.Right(input.Right),
+                EitherStatus.IsLeft => Either<L, R>.Left(input.Left),
+                _ => throw new InvalidOperationException($"Invalid {nameof(EitherStatus)} with value: {input.State}"),
```
Comment (rammba) [inline LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs:16]: Is this correct approach with `InvalidOperationException`, or to use something like

```suggestion
                EitherStatus.IsRight => Either<L, R>.Right(input.Right),
                EitherStatus.IsLeft => Either<L, R>.Left(input.Left),
                _ => Either<L, R>.Bottom,
```

Comment (louthy) [reply LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs:16]: I think the exception approach is best here, as it is an exceptional and unknown value, defaulting to bottom is misleading I think.

File: LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            };
+
+        public static TryAsync<A> ToTryAsync<E, A>(this EitherData<E, A> input) where E : Exception =>
+            input.ToTry().ToAsync();
+
+        public static async Task<TryAsync<A>> ToTryAsync<E, A>(this Task<EitherData<E, A>> input) where E : Exception =>
+            (await input).ToTryAsync();
+
+        public static async Task<TryAsync<A>> ToTryAsync<E, A>(this ValueTask<EitherData<E, A>> input) where E : Exception =>
+            (await input).ToTryAsync();
```
Comment (louthy) [inline LanguageExt.Core/Monads/Alternative Value Monads/Either/Either-Shared/EitherData.Extensions.cs:62]: `.ConfigureAwait(false)`

**Review by louthy — CHANGES_REQUESTED** [review]
Almost there, just a few extras needed.

**Comment by louthy** [comment]
Looks good @rammba , Thanks! 👍

**Pull request merged**